From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch base-11

[patch] dasd: random values in result of BIODASDINFO2.

Use kzalloc to get a zeroed buffer for the structure returned to
user space by the BIODASDINFO2 ioctl. Not all fields are set up,
e.g. the read_devno is missing.
Acked-by: Hannes Reinecke <hare@suse.de>

---

 drivers/s390/block/dasd_ioctl.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff -urpN linux-2.6/drivers/s390/block/dasd_ioctl.c linux-2.6-patched/drivers/s390/block/dasd_ioctl.c
--- linux-2.6/drivers/s390/block/dasd_ioctl.c	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6-patched/drivers/s390/block/dasd_ioctl.c	2006-03-22 11:46:51.000000000 +0100
@@ -403,7 +403,7 @@ dasd_ioctl_information(struct block_devi
 	if (!device->discipline->fill_info)
 		return -EINVAL;
 
-	dasd_info = kmalloc(sizeof(struct dasd_information2_t), GFP_KERNEL);
+	dasd_info = kzalloc(sizeof(struct dasd_information2_t), GFP_KERNEL);
 	if (dasd_info == NULL)
 		return -ENOMEM;
 
@@ -446,8 +446,7 @@ dasd_ioctl_information(struct block_devi
 		memcpy(dasd_info->type, device->discipline->name, 4);
 	else
 		memcpy(dasd_info->type, "none", 4);
-	dasd_info->req_queue_len = 0;
-	dasd_info->chanq_len = 0;
+
 	if (device->request_queue->request_fn) {
 		struct list_head *l;
 #ifdef DASD_EXTENDED_PROFILING
